### PR TITLE
Batch shell

### PR DESCRIPTION
**Pull request by seelabs**
## High Level Overview of Change

This PR adds two patches.

1. A patch to import accounts into the address book from a secret seed.
2. A patch to add a new command to the interactive shell: `load_batch`. This takes a file as an argument and runs the commands in that file as if it were typed into the interactive shell.

### CODE REVIEW

File: slk/repl/repl.py [hunk 1]
```diff
@@ -1032,11 +1032,47 @@ def do_new_account(self: SidechainRepl, line: str) -> None:
...
+        seeds = []
+        index = 0
+        while True:
+            if index == len(args):
+                break
+            if args[index] == '-s':
+                print(
+                    "Error: new_account -s switch must come after the alias argument. Type "
+                    '"help" for help.'
+                )
```
Comment (mvadari) [inline slk/repl/repl.py:1047]: There should be a `return` after this, so it doesn't keep processing after this.

File: slk/repl/repl.py [hunk 2]
```diff
@@ -1032,11 +1032,47 @@ def do_new_account(self: SidechainRepl, line: str) -> None:
...
+                break
+            if args[index] != '-s':
+                continue
+            # user supplied seed
+            index += 1
+            if index == len(args):
+                print(
+                    "Error: new_account -s switch takes one argument. Type "
+                    '"help" for help.'
+                )
```
Comment (mvadari) [inline slk/repl/repl.py:1061]: ditto

File: slk/repl/repl.py [hunk 3]
```diff
@@ -1527,6 +1563,91 @@ def do_setup_accounts(self: SidechainRepl, line: str) -> None:
...
+                    # remove comments. A comment is a '#' and anything that follows it up to the end of a line
+                    # Note this doesn't account for '#' inside strings or escaped, but that should be fine for this
+                    # simple repl. It doesn't have strings anyway.
+                    l = l.split('#')[0].strip()
+                    if not l:
+                        continue
+                    print(f'Running: {l}')
+                    self.onecmd(l)
+        except:
+            print('Error: load_batch command threw an exception')
```
Comment (mvadari) [inline slk/repl/repl.py:1599]: This should also print the original exception in some way, for ease of debugging.

File: slk/repl/repl.py [hunk 4]
```diff
@@ -1527,6 +1563,91 @@ def do_setup_accounts(self: SidechainRepl, line: str) -> None:
...
+                return
+            with open(path) as f:
+                for l in f:
+                    # remove comments. A comment is a '#' and anything that follows it up to the end of a line
+                    # Note this doesn't account for '#' inside strings or escaped, but that should be fine for this
+                    # simple repl. It doesn't have strings anyway.
+                    l = l.split('#')[0].strip()
+                    if not l:
+                        continue
+                    print(f'Running: {l}')
```
Comment (mvadari) [inline slk/repl/repl.py:1596]: nit:
```suggestion
                    print(f'Running: `{l}`')
```

**Review by mvadari — COMMENTED** [review]
Re fixing the linter issues: there are instructions in the CONTRIBUTING doc for how to set up the linter locally.

File: slk/repl/repl.py (hunk 1 above)
Comment (seelabs) [reply slk/repl/repl.py:1047]: Fixed in `491c3e8 fixup! feat: Add load_batch command (#89)`

Comment (seelabs) [reply slk/repl/repl.py:1047]: Fixed in `491c3e8 fixup! feat: Add load_batch command (#89)`

File: slk/repl/repl.py (hunk 2 above)
Comment (seelabs) [reply slk/repl/repl.py:1062]: Fixed in `491c3e8 fixup! feat: Add load_batch command (#89)`

File: slk/repl/repl.py (hunk 3 above)
Comment (seelabs) [reply slk/repl/repl.py]: Fixed in `491c3e8 fixup! feat: Add load_batch command (#89)`

File: slk/repl/repl.py (hunk 4 above)
Comment (seelabs) [reply slk/repl/repl.py]: Fixed in `491c3e8 fixup! feat: Add load_batch command (#89)`

**Review by mvadari — APPROVED** [review]

**Pull request merged**